test/inspector/inspector-helper.js

```diff
@@ -0,0 +1,424 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const http = require('http');
+const path = require('path');
+const spawn = require('child_process').spawn;
+
+const TIMEOUT = 15 * 1000;
+
+const mainScript = path.join(common.fixturesDir, 'loop.js');
+
+function send(socket, message, id, callback) {
+  const msg = JSON.parse(JSON.stringify(message)); // Clone!
+  msg['id'] = id;
+  const messageBuf = Buffer.from(JSON.stringify(msg));
+
+  const wsHeaderBuf = Buffer.allocUnsafe(16);
+  wsHeaderBuf.writeUInt8(0x81, 0);
+  let byte2 = 0x80;
+  const bodyLen = messageBuf.length;
+  let maskOffset = 2;
+  if (bodyLen < 126) {
+    byte2 = 0x80 + bodyLen;
+  } else if (bodyLen < 65536) {
+    byte2 = 0xFE;
+    wsHeaderBuf.writeUInt16BE(bodyLen, 2);
+    maskOffset = 4;
+  } else {
+    byte2 = 0xFF;
+    wsHeaderBuf.writeUInt32BE(bodyLen, 2);
+    wsHeaderBuf.writeUInt32BE(0, 6);
+    maskOffset = 10;
+  }
+  wsHeaderBuf.writeUInt8(byte2, 1);
+  wsHeaderBuf.writeUInt32BE(0x01020408, maskOffset);
+
+  for (let i = 0; i < messageBuf.length; i++)
+    messageBuf[i] = messageBuf[i] ^ (1 << (i % 4));
+  socket.write(
+      Buffer.concat([wsHeaderBuf.slice(0, maskOffset + 4), messageBuf]),
+      callback);
+}
+
+function processIncomingMessage(socket, buffer, handler) {
+  if (buffer.length < 2)
+    return 0;
+  assert.strictEqual(0x81, buffer[0]);
+  let dataLen = 0x7F & buffer[1];
+  let bodyOffset = 2;
+  if (buffer.length < bodyOffset + dataLen)
+    return 0;
+  if (dataLen === 126) {
+    dataLen = buffer.readUInt16BE(2);
+    bodyOffset = 4;
+  } else if (dataLen === 127) {
+    dataLen = buffer.readUInt32BE(2);
+    bodyOffset = 10;
+  }
+  if (buffer.length < bodyOffset + dataLen)
+    return 0;
+  const message = JSON.parse(
+      buffer.slice(bodyOffset, bodyOffset + dataLen).toString('utf8'));
+  handler(message);
+  return bodyOffset + dataLen;
+}
+
+function tearDown(child, err) {
+  child.kill();
+  if (err instanceof Error) {
+    console.error(err.stack);
+    process.exit(1);
+  }
+}
+
+function checkHttpResponse(port, path, callback) {
+  http.get({port, path}, function(r) {
+    let response = '';
+    r.
```

Comment:
Call `r.setEncoding('utf8')` first if you want string data in the data event listener.  (And maybe rename the variable to `res` for clarity.)